compose/service/module.go

```diff
@@ -443,19 +444,46 @@ func (svc module) FindSensitive(ctx context.Context, filter types.PrivacyModuleF
 
 		for _, m := range mm {
 			isPrivate := false
+			// Check module field's sensitivity level
 			for _, f := range m.Fields {
 				if !isPrivate {
-					isPrivate = f.Private
+					isPrivate = f.Privacy.SensitivityLevel > 0
 				}
 			}
 
+			// Check module's sensitivity level
+			if !isPrivate {
+				isPrivate = m.Privacy.SensitivityLevel > 0
+			}
+
+			// Check sensitivity level of connection
+			cMeta, err := svc.dal.GetConnectionMeta(ctx, m.ModelConfig.ConnectionID)
+			if err != nil {
+				return err
+			}
+
+			if !isPrivate {
+				isPrivate = cMeta.SensitivityLevel > 0
+			}
+
 			if isPrivate && m != nil {
-				set = append(set, types.PrivacyModule{
+				pm := types.PrivacyModule{
 					ID:           m.ID,
 					Name:         m.Name, // @todo get this as per translation
 					Handle:       m.Handle,
+					NamespaceID:  m.NamespaceID,
 					ConnectionID: m.ModelConfig.ConnectionID,
-				})
+				}
```

Comment:
What about module fields; do we not want those here?

Speaking of, the `isPrivate` is no longer correct as is since now all modules of a connection would be included (this bit `isPrivate = cMeta.SensitivityLevel > 0`).
Shouldn't we just look at the field's privacy level?

btw, you can replace all those `if`s  with simple bool operations and remove the branching

Reply:
> What about module fields?

No, but it could be beneficial for the future, I will add it. :)

`isPrivate` was misleading and I didn't rename it after sensitive level changes, fixed it now.

Initially, I did only check the field's privacy level, but we are listing sensitive modules, so along with the module's field, If module and connection have sensitive level then shouldn't we consider those too?